src/ripple/core/JobCounter.h

```diff
@@ -159,8 +155,7 @@ class JobCounter
         // construct the wrapped job if we were not previously joined.
         // That would introduce a timing window.
         boost::optional<CountedJob<JobHandler>> ret {
-            CountedJob<JobHandler> {
-                *this, std::forward<JobHandler> (jobHandler)}};
+            CountedJob<JobHandler> {*this, std::move (handler)}};
```

Comment:
After sleeping on it, I'm having second thoughts about using `std::move` here instead of `std::forward<>`.  They both do the right thing in this context.  The `std::move` is a little easier to read in the current situation.  But if, in the future, we were to add support for `lvalue lambdas` by removing the `static_assert`, then the `std::move` would be wrong (and easy to miss).  Therefore I'm inclined to think  `std::forward<>` is a better choice in terms of maintenance.  However it's only a weak inclination.

Thoughts?